include/anvill/Arch.h

```diff
@@ -0,0 +1,186 @@
+struct RegisterConstraint {
+  RegisterConstraint(std::vector<VariantConstraint> _variants)
+      : variants(_variants) {}
+
+  std::vector<VariantConstraint> variants;
+};
+
+class CallingConvention {
+ public:
+  CallingConvention(llvm::CallingConv::ID _identity) : identity(_identity) {}
+  virtual ~CallingConvention(){};
+
+  virtual std::vector<ParameterDecl> BindParameters(
+      const llvm::Function &function) = 0;
+  virtual std::vector<ValueDecl> BindReturnValues(
+      const llvm::Function &function) = 0;
+  virtual remill::Register *BindReturnStackPointer(
+      const llvm::Function &function) = 0;
+
+  llvm::CallingConv::ID getIdentity() { return identity; }
+
+ private:
+  llvm::CallingConv::ID identity;
+};
+
+class X86_64_SysV : public CallingConvention {
+ public:
+  X86_64_SysV() : CallingConvention(llvm::CallingConv::X86_64_SysV) {}
+  ~X86_64_SysV(){};
+  std::vector<ParameterDecl> BindParameters(const llvm::Function &function);
+  std::vector<ValueDecl> BindReturnValues(const llvm::Function &function);
+  remill::Register *BindReturnStackPointer(const llvm::Function &function);
+
+ private:
+  // TODO: add the rest of the Registers here, not just the 32-bit and 64-bit
+  // variants
+  const std::vector<RegisterConstraint> parameter_register_constraints = {
+      RegisterConstraint({
+          VariantConstraint("EDI", kTypeIntegral, kMaxBit32),
+          VariantConstraint("RDI", kTypeIntegral, kMaxBit64),
+      }),
+      RegisterConstraint({
+          VariantConstraint("ESI", kTypeIntegral, kMaxBit32),
+          VariantConstraint("RSI", kTypeIntegral, kMaxBit64),
+      }),
+      RegisterConstraint({
+          VariantConstraint("EDX", kTypeIntegral, kMaxBit32),
+          VariantConstraint("RDX", kTypeIntegral, kMaxBit64),
+      }),
+      RegisterConstraint({
+          VariantConstraint("ECX", kTypeIntegral, kMaxBit32),
+          VariantConstraint("RCX", kTypeIntegral, kMaxBit64),
+      }),
+      RegisterConstraint({
+          VariantConstraint("R8D", kTypeIntegral, kMaxBit32),
+          VariantConstraint("R8", kTypeIntegral, kMaxBit64),
+      }),
+      RegisterConstraint({
+          VariantConstraint("R9D", kTypeIntegral, kMaxBit32),
+          VariantConstraint("R9", kTypeIntegral, kMaxBit64),
+      }),
+
+      RegisterConstraint({VariantConstraint("XMM0", kTypeFloat, kMaxBit128)}),
+      RegisterConstraint({VariantConstraint("XMM1", kTypeFloat, kMaxBit128)}),
+      RegisterConstraint({VariantConstraint("XMM2", kTypeFloat, kMaxBit128)}),
+      RegisterConstraint({VariantConstraint("XMM3", kTypeFloat, kMaxBit128)}),
+      RegisterConstraint({VariantConstraint("XMM4", kTypeFloat, kMaxBit128)}),
+      RegisterConstraint({VariantConstraint("XMM5", kTypeFloat, kMaxBit128)}),
+      RegisterConstraint({VariantConstraint("XMM6", kTypeFloat, kMaxBit128)}),
+      RegisterConstraint({VariantConstraint("XMM7", kTypeFloat, kMaxBit128)}),
+  };
```

Comment:
Can `XMM8` et al. be used?

Reply:
No, don't think so: https://godbolt.org/z/7w-NLA